src/files/modals/add-by-path-modal/AddByPathModal.js

```diff
@@ -1,52 +1,130 @@
 import React from 'react'
 import PropTypes from 'prop-types'
-import isIPFS from 'is-ipfs'
+import Button from '../../../components/button/Button'
+import { Modal, ModalActions, ModalBody } from '../../../components/modal/Modal'
 import { withTranslation } from 'react-i18next'
+import isIPFS from 'is-ipfs'
 import Icon from '../../../icons/StrokeDecentralization'
-import TextInputModal from '../../../components/text-input-modal/TextInputModal'
 
-function ByPathModal ({ t, tReady, onCancel, onSubmit, className, ...props }) {
-  const validatePath = (p) => {
+const FILENAME_REGEX = /\?filename=(.+)/
+
+class AddByPathModal extends React.Component {
+  static propTypes = {
+    onCancel: PropTypes.func.isRequired,
+    onSubmit: PropTypes.func.isRequired,
+    className: PropTypes.string
+  }
+
+  static defaultProps = {
+    className: ''
+  }
+
+  state = {
+    path: '',
+    name: ''
+  }
+
+  validatePath = (p) => {
     if (!p.startsWith('/ipfs/')) {
       p = `/ipfs/${p}`
     }
 
     return isIPFS.ipfsPath(p)
   }
 
-  const getDescription = () => {
+  onChange = (event) => {
+    const target = event.target
+    const value = target.value.trim()
+    const name = target.name
+
+    if (name === 'path') {
+      const matches = value.match(FILENAME_REGEX)
+      if (matches && matches[1]) {
+        this.setState({ name: matches[1] })
+      }
+    }
```

Comment:
@lidel not particularly happy about this bit to detect the `?filename=` part. Any suggestion?

Reply:
I played a bit with user flow related to this and realized it is not worth headache – decided to remove it in 6b86273